services/replicator/metadataReconciler.go

```diff
@@ -303,9 +303,9 @@ func (r *metadataReconciler) reconcileCg(localCgs []*shared.ConsumerGroupDescrip
 		})
 		localCg, ok := localCgsSet[remoteCg.GetConsumerGroupUUID()]
 		if ok {
-			if remoteCg.GetStatus() == shared.ConsumerGroupStatus_DELETED {
+			if remoteCg.GetStatus() == shared.ConsumerGroupStatus_DELETING {
```

Comment:
I think we should check for both condition here. If remote is in deleting or deleted, and local is not in either of the state, we should call deleteConsumergroup.

Reply:
CGs in "DELETED" state are *not* returned by ListConsumerGroups .. so this code was dead code earlier. It however does return CGs in "DELETING" state .. and the code below that calls "DeleteConsumerGroup" locally would then move the local CG to DELETING, as well.